app/models/ems_refresh/save_inventory.rb

```diff
@@ -1,11 +1,19 @@
 module EmsRefresh::SaveInventory
-  # Parsed inventory can come as hash of hashes or array of InventoryCollection's.
-  def save_ems_inventory(ems, hashes_or_collections, target = nil, disconnect = true)
-    if hashes_or_collections.kind_of?(Array)
-      ManagerRefresh::SaveInventory.save_inventory(ems, hashes_or_collections) # InventoryCollections.
+  # Parsed inventory can come as hash of hashes or array of InventoryCollection's or
+  # ManagerRefresh::Inventory::Persister object.
+  def save_ems_inventory(ems, hashes_or_collections_or_persister, target = nil, disconnect = true)
+    # TODO(lsmola) cleanup all providers so Graph Refresh saving does not go this path at all, then delete both Graph
+    # Refresh related branches
+    if hashes_or_collections_or_persister.kind_of?(ManagerRefresh::Inventory::Persister)
+      persister = hashes_or_collections_or_persister
+      persister.persist!
+      return
+    elsif hashes_or_collections_or_persister.kind_of?(Array)
+      inventory_collections = hashes_or_collections_or_persister
+      ManagerRefresh::SaveInventory.save_inventory(ems, inventory_collections)
```

Comment:
Wow no this is way too complicated :) we just brought this down by one condition in 76dd5cf689237792b0337a9cfd3d21114218b0c3

If we need to change callers to not go down this path lets do that first instead of adding a bunch of if/elsif/else here

Reply:
right, as the TODOon line 5  implies, we can't drop this, until all the providers have changed code. And we can't change the providers before having this PR first.

So this PR brings the new way, while not breaking all the providers in the process :-)